script.js

```diff
@@ -373,366 +373,424 @@ function keyReleased() {
-        curBlock[1].xPos -= 20
-        curBlock[1].yPos -= 20
-        curBlock[3].xPos += 20
-        curBlock[3].yPos += 20
-      } else if (curBlock[0].yPos < curBlock[3].yPos) {
-        curBlock[0].xPos -= 40
-        curBlock[1].xPos += 20
-        curBlock[1].yPos += 20
-        curBlock[3].xPos -= 20
-        curBlock[3].yPos -= 20
-      } else if (curBlock[0].xPos > curBlock[3].xPos) {
-        curBlock[0].yPos -= 40
-        curBlock[1].xPos -= 20
-        curBlock[1].yPos += 20
-        curBlock[3].xPos += 20
-        curBlock[3].yPos -= 20
-      }
-    }
+function checkJRotationCounterClockwise(){
+  if (curBlock[0].xPos <= curBlock[3].xPos && curBlock[0].yPos <= curBlock[3].yPos) {
+    rotateJC3JCC1();
+  } else if (curBlock[0].xPos <= curBlock[3].xPos && curBlock[0].yPos >= curBlock[3].yPos) {
+    rotateJC1JCC2();
+  } else if (curBlock[0].xPos >= curBlock[3].xPos && curBlock[0].yPos <= curBlock[3].yPos) {
+    rotateJC4JCC3();
+  } else if (curBlock[0].xPos >= curBlock[3].xPos && curBlock[0].yPos >= curBlock[3].yPos) {
+    rotateJC2JCC4();
   }
+}
 
-  if (curBlock[0].type == "S") {
-    if (direction == 1) {
-      if (curBlock[0].xPos < curBlock[3].xPos) {
-        curBlock[1].xPos += 40
-        curBlock[0].xPos += 20
-        curBlock[0].yPos -= 20
-        curBlock[2].xPos -= 20
-        curBlock[2].yPos -= 20
-      } else if (curBlock[0].yPos > curBlock[3].yPos) {
-        curBlock[1].yPos -= 40
-        curBlock[0].xPos -= 20
-        curBlock[0].yPos -= 20
-        curBlock[2].xPos -= 20
-        curBlock[2].yPos += 20
-      } else if (curBlock[0].yPos < curBlock[3].yPos) {
-        curBlock[1].yPos += 40
-        curBlock[0].xPos += 20
-        curBlock[0].yPos += 20
-        curBlock[2].xPos += 20
-        curBlock[2].yPos -= 20
-      } else if (curBlock[0].xPos > curBlock[3].xPos) {
-        curBlock[1].xPos -= 40
-        curBlock[0].xPos -= 20
-        curBlock[0].yPos += 20
-        curBlock[2].xPos += 20
-        curBlock[2].yPos += 20
-      }
-    }
-    if (direction == 2) {
-      if (curBlock[0].xPos < curBlock[3].xPos) {
-        curBlock[1].yPos += 40
-        curBlock[0].xPos += 20
-        curBlock[0].yPos += 20
-        curBlock[2].xPos += 20
-        curBlock[2].yPos -= 20
-      } else if (curBlock[0].yPos > curBlock[3].yPos) {
-        curBlock[1].xPos += 40
-        curBlock[0].xPos += 20
-        curBlock[0].yPos -= 20
-        curBlock[2].xPos -= 20
-        curBlock[2].yPos -= 20
-      } else if (curBlock[0].yPos < curBlock[3].yPos) {
-        curBlock[1].xPos -= 40
-        curBlock[0].xPos -= 20
-        curBlock[0].yPos += 20
-        curBlock[2].xPos += 20
-        curBlock[2].yPos += 20
-      } else if (curBlock[0].xPos > curBlock[3].xPos) {
-        curBlock[1].yPos -= 40
-        curBlock[0].xPos -= 20
-        curBlock[0].yPos -= 20
-        curBlock[2].xPos -= 20
-        curBlock[2].yPos += 20
-      }
-    }
+function checkJRotationDirection(bool){
```

Comment:
It's a good idea to give booleans good names. If you changed `bool` to `clockwise` this would be super easy to read and see that it's correct. Also someone looking at the function signature would know that a variable named `clockwise` should be given `true` for clockwise and `false` for counter clockwise. With `bool` they have to read the code to see which value is which.